### ISSUE TITLE
Remove NoInline compiler rule when TryCatch optimization is shipped in Turbofan

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#4104]
Currently Inline TryCatch statements deoptimize the entire function containing them. https://github.com/ampproject/amphtml/pull/4028 was created to move TryCatch out of a few functions and a new rule (if name ends with "NoInline") was added to Closure Compiler to ensure CC does not inline them again at compile time. 

We should remove this rule after V8 ships the TryCatch optimization for Turbofan. (It was [shipped](https://codereview.chromium.org/1713033002) and then [unshipped](https://codereview.chromium.org/1931973002) due to regression)

**Comment by aghassemi** [comment]
https://codereview.chromium.org/1996373002 is the latest attempt to ship it. So it might be finally in soon

**Comment by aghassemi** [comment]
Based on the discussion here: https://groups.google.com/forum/#!topic/v8-users/maH60gh_a8s https://codereview.chromium.org/1996373002 is likely to ship. So we should be able to test and remove this in a month or two.

**Comment by adelinamart** [comment]
Is this completed? Thanks.

**Comment by aghassemi** [comment]
Not completed yet and low priority.

**Comment by adelinamart** [comment]
@aghassemi if this is 'framework', shall we assign to @jridgewell ? Thanks

**Comment by aghassemi** [comment]
@jridgewell would be happy when these are gone :) either way is fine with me.

**Comment by sparhami** [comment]
Its been a while since I have seen the yellow warning triangle in dev tools for a de-optimized function for try/catch. I ran a jsperf I found to play with this: https://jsperf.com/try-catch-performance-overhead

I first ran it in Chrome 5 (to make sure the jsperf itself was good):

![screenshot from 2018-06-21 15-31-55](https://user-images.githubusercontent.com/1410321/41748821-4d76daee-7568-11e8-87c5-84e1f0d91531.png)

And then again in a Chrome 67:

![screenshot from 2018-06-21 15-40-18](https://user-images.githubusercontent.com/1410321/41749086-6f6e7a5c-7569-11e8-88d3-1bdf6610e7f2.png)

I'm not sure if everything related to the `try`/`catch` optimization is in, but I think it should at least be in some state of working these days. Might be worth doing a bit more investigation.
